### PR TITLE
Document ANSI-compliant trim function

### PR DESCRIPTION
**Pull request by m57lyra**
## Description

Add entry in string.rst for new ANSI-compliant trim() function from #11347

> Is this change a fix, improvement, new feature, refactoring, or other?
It is documentation for #11347 

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)
No.

> How would you describe this change to a non-technical end user or system administrator?
It is documentation.

## Documentation

( ) No documentation is needed.
(x) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: docs/src/main/sphinx/functions/string.rst [hunk 1]
```diff
@@ -248,7 +248,7 @@ String functions
        SELECT translate('abcd', 'a', 'z'); -- 'zbcd'
        SELECT translate('abcda', 'a', 'z'); -- 'zbcdz'
        SELECT translate('Palhoça', 'ç','c'); -- 'Palhoca'
-       SELECT translate('abcd', 'b', U&'\+01F600'); -- a😀cd
```
Comment (martint) [inline docs/src/main/sphinx/functions/string.rst:251]: Unrelated change, please revert.

Comment (m57lyra) [reply docs/src/main/sphinx/functions/string.rst:251]: On it. Thought it was a bug!

File: docs/src/main/sphinx/functions/string.rst [hunk 2]
```diff
@@ -258,6 +258,17 @@ String functions
 
     Removes leading and trailing whitespace from ``string``.
 
+.. function:: trim( [ [ {LEADING | TRAILING | BOTH} ] [ string ] FROM ]
+                    charValueExpression ) -> varchar
```
Comment (ebyhr) [inline docs/src/main/sphinx/functions/string.rst:262]: How about using simple declaration and mention the details in body? e.g. 
```
.. function:: trim(specification trim FROM string) -> varchar
```

File: docs/src/main/sphinx/functions/string.rst [hunk 3]
```diff
@@ -258,6 +258,17 @@ String functions
...
+
+    Removes specified whitespace or string from ``charValueExpression`` at
+    ``string``, where ``charValueExpression`` can be a quoted string
+    or a function::
```
Comment (ebyhr) [inline docs/src/main/sphinx/functions/string.rst:266]: I think there's no need to mention that the argument accepts functions. Existing functions don't have such explanation.

File: docs/src/main/sphinx/functions/string.rst [hunk 4]
```diff
@@ -258,6 +258,17 @@ String functions
...
+
+      SELECT trim( LEADING FROM '  abcd' );  -- 'abcd'
+      SELECT trim( BOTH '$' FROM '$var$'); -- 'var'
+      SELECT trim( TRAILING 'ER' FROM upper('worker') ); -- 'WORK'
```
Comment (ebyhr) [inline docs/src/main/sphinx/functions/string.rst:270]: nit: Remove redundant spaces.
```suggestion
      SELECT trim(LEADING FROM '  abcd'); -- 'abcd'
      SELECT trim(BOTH '$' FROM '$var$'); -- 'var'
      SELECT trim(TRAILING 'ER' FROM upper('worker')); -- 'WORK'
```

File: docs/src/main/sphinx/functions/string.rst [hunk 5]
```diff
@@ -258,6 +258,17 @@ String functions
...
 
+.. function:: trim( [ [ {LEADING | TRAILING | BOTH} ] [ string ] FROM ]
+                    charValueExpression ) -> varchar
+
+    Removes specified whitespace or string from ``charValueExpression`` at
+    ``string``, where ``charValueExpression`` can be a quoted string
+    or a function::
+
+      SELECT trim( LEADING FROM '  abcd' );  -- 'abcd'
+      SELECT trim( BOTH '$' FROM '$var$'); -- 'var'
```
Comment (ebyhr) [inline docs/src/main/sphinx/functions/string.rst:269]: Could you add another example without specification? e.g.
```sql
SELECT trim('!' FROM '!foo!'); -- 'foo'
```

**Review by mosabua — APPROVED** [review]
Looks great now.
